### ISSUE TITLE
[Question] React-leaflet does not let react-modal close on first click

### ISSUE THREAD

**Issue by 18anandn** [PaulLeCam/react-leaflet#1103]
## Bug report in v4

- [X] All peer dependencies are installed: React, ReactDOM and Leaflet.
- [X] Using the latest version of React and ReactDOM v18.
- [X] Using the supported version of Leaflet (v1.8.0 minimum) and its corresponding CSS file is loaded.
- [X] Using the [latest v4 version of React-Leaflet](https://github.com/PaulLeCam/react-leaflet/releases).
- [X] The issue has not already been reported.
- [X] Make sure you have followed the [quick start guide](https://leafletjs.com/examples/quick-start.html) for Leaflet.
- [X] Make sure you have fully read the [documentation](https://react-leaflet.js.org/docs/start-introduction) and that you understand the [limitations](https://react-leaflet.js.org/docs/start-introduction#limitations).

### Expected behavior

After clicking on the zoom button, modal closes when clicked on overlay on first click.

### Actual behavior

After clicking on the zoom button, modal does not close when clicked on overlay on first click. It takes 2 clicks. It works as expected with every other interaction (like click, scroll zoom).

### Steps to reproduce

Dependencies:
`
{
        "leaflet": "^1.9.4",
        "react": "^18.2.0",
        "react-dom": "^18.2.0",
        "react-leaflet": "^4.2.1",
        "react-modal": "^3.16.1"
}
`

Code to reproduce:
`
import { useState } from 'react';
import ReactModal from 'react-modal';
import { MapContainer } from 'react-leaflet/MapContainer';
import { TileLayer } from 'react-leaflet/TileLayer';

function App() {
  const [isOpenModal, setIsOpenModal] = useState<boolean>(false);

  const handleClick = () => {
    setIsOpenModal(true);
  };

  return (
    <>
      <button onClick={handleClick}>Open React Leaflet</button>
      <ReactModal
        isOpen={isOpenModal}
        onRequestClose={() => setIsOpenModal(false)}
        parentSelector={() => document.body}
      >
        <MapContainer
          center={[51.505, -0.09]}
          zoom={13}
          scrollWheelZoom={false}
          style={{
            height: '50dvh',
            width: '50dvw',
          }}
        >
          <TileLayer
            attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
          />
        </MapContainer>
      </ReactModal>
    </>
  );
}

export default App;
`

Please provide the simplest example possible to reproduce the issue, based on [this StackBlitz](https://stackblitz.com/edit/react-leaflet-v4?file=src/App.js).

**Comment by cowglow** [comment]
When you say the Zoom button, do you mean from the map controls? Because there isn't anything in your code snippet that would trigger that behavior.  You might want to use the useMapEvent and listen for the end of the Zoom to trigger the stage change for the modal to close.

**Comment by 18anandn** [comment]
> When you say the Zoom button, do you mean from the map controls? Because there isn't anything in your code snippet that would trigger that behavior. You might want to use the useMapEvent and listen for the end of the Zoom to trigger the stage change for the modal to close.

@cowglow 

Yes I mean the zoom button provided by react-leaflet by default. If I use touchpad or keyboard (ctrl and +) to zoom then this problem does not arise.
